shell.go

```diff
@@ -84,6 +84,42 @@ func NewShellWithClient(url string, c *gohttp.Client) *Shell {
 	}
 }
 
+// Get shell from environmental variables, default api path or gateway.
```

Comment:
We should spell out exactly how we pick the default shell.

Reply:
What do you think if I add some functions which respectively obtain new shells from environmental variables, default gateway and the api file. I think the order of shells to try in the function DefautlShell is sensible, as normally the environmental variables are explicitly set by the user and I have tried to use gateway as the the last resort.

Reply:
Sounds good to me.